src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -422,7 +520,7 @@ protected ODataJsonLightReaderNestedResourceInfo InnerReadUndeclaredProperty(IOD
 
                 // Complex property or collection of complex property.
                 bool isCollection = payloadTypeReference.IsCollection();
-                ValidateExpandedNestedResourceInfoPropertyValue(this.JsonReader, isCollection, propertyName, payloadTypeReference);
+                ValidateExpandedNestedResourceInfoPropertyValue(this.JsonReader.NodeType, () => this.JsonReader.Value, isCollection, propertyName, payloadTypeReference);

```

Comment:
Could you explain the reasoning behind these changes? Why is it better to pass the node type and value delegate separately instead of passing the `JsonReader` itself?

Also, note that `() => this.JsonReader.Value` will cause a new Func to be allocated  each time the method  is called as [demonstrated here](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLRIMYHsEBNXIwJwYzIA+AAgEwCMAsAFCUDMABDWwMJsDeTbQR3YAlaLggI2AWQCeYqBIQBuAULWDWbYFiwAbNgtwAKAJR8NQwVmAArCKTYA3KHrgQ2AXjYwAFgEtkADoANVd/XFgIYzMvAD4fAOC5I0lQ13dTVUYrK0oAdmcMjwBCbwA7OD09bKsAX0tLLRt7RzC9CKjjSgBWAB4WhxgElzcIABEIPQgAcyjzfhzczULR90npuZhorMsGxn2mLU5UqUW89kG24r42GYgYZTZkB6f9uqA=)

Reply:
Fixed by removing the delegate